### PR TITLE
FBX Conversion Crash Fix

### PR DESCRIPTION
**Pull request by DamianMadden**
Resolves [AB#575](https://dev.azure.com/euclideon/57c3bd2a-8f94-4578-b195-2e4fa9d8f295/_workitems/edit/575)

Resolves [AB#779](https://dev.azure.com/euclideon/57c3bd2a-8f94-4578-b195-2e4fa9d8f295/_workitems/edit/779)

### CODE REVIEW

File: src/vcFBX.cpp [hunk 1]
```diff
@@ -357,6 +358,9 @@ vdkError vcFBX_Open(vdkConvertCustomItem *pConvertInput, uint32_t everyNth, cons
   pFBX->lastTouchedPoly = 1; // Forces handling 0, then gets set to 0
   pFBX->lastTouchedMesh = 1;
 
+  if (pFBX->pTrivox != nullptr)
```
Comment (samsinsane) [inline src/vcFBX.cpp]: This check is already being done within `vdkTriangleVoxelizer_Destroy`

File: src/vcFBX.cpp [hunk 2]
```diff
@@ -365,7 +369,7 @@ vdkError vcFBX_Open(vdkConvertCustomItem *pConvertInput, uint32_t everyNth, cons
 
   if (result != vE_Success)
   {
-    pFBX->pManager->Destroy(); // Destroying manager destroys all objects that were created with it
+    pFBX->pManager->Destroy();
```
Comment (samsinsane) [inline src/vcFBX.cpp]: Is this comment no longer true?

File: src/vcFBX.cpp (hunk 1 above)
Comment (peterAdams1014) [reply src/vcFBX.cpp]: This change is actually unnecessary, as `pFBX->pTrivox` is always `nullptr` at this point (`vcFBX_Close()` destroys it).

File: src/vcFBX.cpp [hunk 3]
```diff
@@ -804,8 +806,43 @@ void vcFBX_Close(vdkConvertCustomItem *pConvertInput)
     if (pFBX->pTrivox)
       vdkTriangleVoxelizer_Destroy(&pFBX->pTrivox);
 
+    if (pFBX->pScene != nullptr)
```
Comment (peterAdams1014) [inline src/vcFBX.cpp]: Unnecessary check, `pFBX->pScene` will be allocated here.

File: src/vcFBX.cpp [hunk 4]
```diff
@@ -804,8 +806,43 @@ void vcFBX_Close(vdkConvertCustomItem *pConvertInput)
...
+    {
+      pFBX->pScene->Destroy(true);
+      pFBX->pScene = nullptr;
+    }
+
     if (pFBX->pManager != nullptr)
```
Comment (peterAdams1014) [inline src/vcFBX.cpp]: Unnecessary check, `pFBX->pManager` will be allocated here.

File: src/vcFBX.cpp [hunk 5]
```diff
@@ -804,8 +806,43 @@ void vcFBX_Close(vdkConvertCustomItem *pConvertInput)
     if (pFBX->pTrivox)
```
Comment (peterAdams1014) [inline src/vcFBX.cpp]: Unnecessary check, `pFBX->pTrivox` will be allocated here.

File: src/vcFBX.cpp [hunk 6]
```diff
@@ -804,8 +806,43 @@ void vcFBX_Close(vdkConvertCustomItem *pConvertInput)
...
+      pFBX->pManager = nullptr;
+    }
+
+    pFBX->uvQueue.Deinit();
+
+    vcFBX_CleanMaterials(&pFBX->materials);
+  }
+}
+
+void vcFBX_Destroy(vdkConvertCustomItem* pConvertInput)
```
Comment (peterAdams1014) [inline src/vcFBX.cpp]: `vcFBX_Destroy()` will be called after `vcFBX_Close` always. The only thing that should be be cleaned up here is anything allocated inside `vcFBX_AddItem()`.

Checking `pConvertInput->pData != nullptr` is also unnecessary.

File: src/vcFBX.cpp (hunk 2 above)
Comment (DamianMadden) [reply src/vcFBX.cpp]: Never was quite true I believe that comment was leftover from johnathon

File: src/vcFBX.cpp (hunk 1 above)
Comment (DamianMadden) [reply src/vcFBX.cpp]: Oh yeah I got kind of heavy handed when stuff was still leaking after innumerable changes, will do another pass

**Comment by OsiJr** [comment]
Was closed accidentally because target branch was merged and deleted.

**Pull request merged**